api/traces/v1/http.go

```diff
@@ -163,6 +165,47 @@ func NewV2Handler(read *url.URL, opts ...HandlerOption) http.Handler {
 	return r
 }
 
+// Parse a URL; if the URL includes `{tenant}` that portion will be replaced by the tenant
+func ExpandTemplatedUpstream(templateUpstream, tenant string) (*url.URL, error) {
+	rawTracesReadEndpoint := strings.Replace(templateUpstream, "{tenant}", tenant, 1)
```

Comment:
Nothing; and this is by design, mostly for testing.

I thought about validating that the parameter `--experimental.traces.read.endpoint-template` includes `{tenant}`, as the user will not get the behavior without including it.

However, if I allow the template to be a constant, I can test the code without DNS manipulation.

So I would like to keep the behavior of not requiring a template.  Possibly we could add a warning message, if you think it would help the deployer.

Reply:
Lets add a warning message then :+1: